### PR TITLE
fix: search-oidc-config

### PR DESCRIPTION
**Pull request by petehanssens**
### Summary of Changes

OIDC config added to Dockerfile.search.public

### Tests

I manually built the image and tested it in my own environment

### Documentation

None but will be writing a blog post on okta OIDC setup

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: Dockerfile.search.public [hunk 1]
```diff
@@ -8,3 +8,21 @@ COPY requirements-common.txt /app/requirements-common.txt
...
+
+ENV FLASK_APP_MODULE_NAME flaskoidc
+ENV FLASK_APP_CLASS_NAME FlaskOIDC
+ENV FLASK_OIDC_WHITELISTED_ENDPOINTS status,healthcheck,health
+ENV SQLALCHEMY_DATABASE_URI sqlite:///sessions.db
+
+# You will need to set these environment variables in order to use the oidc image
+# FLASK_OIDC_CLIENT_SECRETS - a path to a client_secrets.json file
+# FLASK_OIDC_SECRET_KEY - A secret key from your oidc provider
+# You will also need to mount a volume for the clients_secrets.json file.
```
Comment (verdan) [inline Dockerfile.search.public:25]: This is not true.

File: Dockerfile.search.public [hunk 2]
```diff
@@ -8,3 +8,21 @@ COPY requirements-common.txt /app/requirements-common.txt
...
+
+RUN  pip3 install -e .&& \
+     pip3 install -e .[oidc]
+
+ENV FLASK_APP_MODULE_NAME flaskoidc
+ENV FLASK_APP_CLASS_NAME FlaskOIDC
+ENV FLASK_OIDC_WHITELISTED_ENDPOINTS status,healthcheck,health
+ENV SQLALCHEMY_DATABASE_URI sqlite:///sessions.db
+
+# You will need to set these environment variables in order to use the oidc image
```
Comment (verdan) [inline Dockerfile.search.public:22]: You also need to add `FLASK_OIDC_CONFIG_URL`
Would be nice to add this link for ref: https://github.com/verdan/flaskoidc#readme

This change should go across all repos.

**Review by verdan — CHANGES_REQUESTED** [review]

File: Dockerfile.search.public (hunk 2 above)
Comment (petehanssens) [reply Dockerfile.search.public:22]: @verdan let me know if this is resolved

File: Dockerfile.search.public (hunk 1 above)
Comment (petehanssens) [reply Dockerfile.search.public]: @verdan i've removed

**Review by mgorsk1 — CHANGES_REQUESTED** [review]
thanks for contribution @petehanssens! I think it would also be useful to extend https://github.com/amundsen-io/amundsen/blob/main/search/Makefile to build both regular and oidc images (just like we do for metadata and frontend, example here: https://github.com/amundsen-io/amundsen/blob/main/frontend/Makefile#L52-L54).

This way oidc image will be build by GH Action and pushed to dockerhub on release.

Also - remember to complete DCO so our checks are happy. Thanks!

**Comment by petehanssens** [comment]
Hi @verdan I think this branch has gotten a bit stale so it might be worthwhile closing

**Pull request closed without merging**